utils/location.go

```diff
@@ -163,7 +163,12 @@ func (h *LocationHierarchy) FindByName(name string, level LocationLevel, parent
 
 // FindByPath looks for a location in the hierarchy with the given path
 func (h *LocationHierarchy) FindByPath(path string) *Location {
-	return h.pathLookup.lookup(strings.ToLower(path))
+	tokens := strings.Split(path, ">")
+	for i := range tokens {
+		tokens[i] = strings.TrimSpace(tokens[i])
```

Comment:
I think we can do a little bit more normalization here of each location token within the path
  1. trim whitespace
  2. replace `\w+` with a single space
  3. lowercase (don't need to do it again below)

so ` Kigali\x20\t\x20City ` becomes `kigali city`